spec/db.spec.ts

```diff
@@ -94,6 +98,96 @@ describe('db', () => {
 
           return expect(dbConn.connect()).to.not.be.rejected;
         })
+
+        // Skip ssh tests for adapters that have it disabled
+        const describeSSH = dbAdapterSettings?.disabledFeatures.some((f) => f === 'server:ssh')
+          ? describe.skip
+          : describe;
+
+        describeSSH('connect with ssh', () => {
+          const sshHost = process.env.SSH_HOST || 'localhost';
+
+          const assertSSHConnection = async (dbConn: Database) => {
+            await dbConn.connect();
+
+            // assert beyond a simplly opening the connection,
+            // but also ensure it can run a query
+            const databases = await dbConn.listDatabases();
+            expect(databases.length).to.be.above(0);
+          };
+
+          it('should connect into server using ssh with password', () => {
+            const serverInfo: ServerConfig = {
+              ...config[dbAdapter],
+              name: dbAdapter,
+              adapter: dbAdapter,
+            };
+
+            serverInfo.ssh = {
+              host: sshHost,
+              port: 2222,
+              user: 'sqlectron',
+              password: 'password',
+            };
+
+            const serverSession = db.createServer(serverInfo);
+            const dbConn = serverSession.createConnection(database);
+
+            return expect(assertSSHConnection(dbConn)).to.not.be.rejected;
+          });
+
+          const keyTypes = ['rsa', 'ecdsa', 'ed25519']
+
+          for (const keyType of keyTypes) {
+            it(`should connect into server using ssh with private key of type ${keyType}`, () => {
+              const serverInfo: ServerConfig = {
+                ...config[dbAdapter],
+                name: dbAdapter,
+                adapter: dbAdapter,
+              };
+
+              serverInfo.ssh = {
+                host: sshHost,
+                port: 2222,
+                user: 'sqlectron',
+                privateKey: path.join(__dirname, 'ssh_files/id_' + keyType),
+              };
+
+              const serverSession = db.createServer(serverInfo);
+              const dbConn = serverSession.createConnection(database);
+
+              // ed25519 is only supported by node v12+
+              if (keyType === 'ed25519' && (process.version.startsWith('v10') || process.version.startsWith('v8'))) {
```

Comment:
```suggestion
              if (keyType === 'ed25519' && process.version.startsWith('v10')) {
```

Minimum supported version for this package is 10.13, so don't need to worry about the test suite running on 8.x:
https://github.com/sqlectron/sqlectron-db-core/blob/0369c8a02b47f5c1ab087ace98c4f731eab84712/package.json#L35-L37